sched/tls/task_initinfo.c

```diff
@@ -28,6 +28,8 @@
 #include <nuttx/kmalloc.h>
 #include <nuttx/mutex.h>
 
+#include <nuttx/lib/lib.h>
```

Comment:
change to libc.h

Reply:
Yes, forget it's in side kernel space, but kernel code can't reference libc function directly even th inclusion is added. The better fix should be that seek/fwrite/fread call lseek/_NX_WRITE/_NX_READ directly if the related callback is NULL.

Reply:
@michallenc this is the last comment you need consider, otherwise the change will break protected and kernel mode.

Reply:
> `<nuttx/lib/lib.h>` was used to reference `lib_stream_initialize` and `lib_stream_release` before your patch so I am not sure if it is such a problem in this case.

both functions are moved into sched, so the inclusion doesn't need anymore

> Deciding what function is to be called would require some if statements in running code which I would like to avoid.

but you have to do th check since:

1. stdin/stdout/sderr is initialized inside kernel space
2. The assignment of stdin/stdout/stderr's callback inside kernel space point to the lib_fxxx_cb inside kernel space
3. fopen/fdopen point lib_fxxx_cb in user space

when user call fwrite/fread on stdin/stdout/stderr, prefetch abort will happen since user space doesn't right to execute the code inside the kernel space.